tagstudio/src/qt/modals/tag_database.py

```diff
@@ -54,23 +56,25 @@ def __init__(self, library: Library):
 
         self.root_layout.addWidget(self.search_field)
         self.root_layout.addWidget(self.scroll_area)
-        self.update_tags()
+        if self.lib.storage_path:
+            self.update_tags()
 
     def on_return(self, text: str):
-        if text and self.first_tag_id >= 0:
-            # callback(self.first_tag_id)
-            self.search_field.setText("")
-            self.update_tags()
+        if text:
+            # self.search_field.setText("")
+            self.update_tags(text)
         else:
             self.search_field.setFocus()
-            self.parentWidget().hide()
+            if self.is_popup:
+                self.parentWidget().hide()
 
     def update_tags(self, query: str | None = None):
         # TODO: Look at recycling rather than deleting and re-initializing
+        logger.info("updating tags", query=query)
         while self.scroll_layout.itemAt(0):
             self.scroll_layout.takeAt(0).widget().deleteLater()
 
-        tags = self.lib.search_tags(FilterState(path=query, page_size=self.tag_limit))
+        tags = self.lib.search_tags(FilterState(tag=query, page_size=self.tag_limit))
```

Comment:
@cyanvoxel you might want to fix this in upstream as well - the search query should be `tag` instead of `path` here